dcp/DcpHandler.go

```diff
@@ -524,25 +531,32 @@ func (m *Mutation) ToUprEvent() *xdcrBase.WrappedUprEvent {
 //	collectionId - 4 bytes
 //	colFiltersLen - 2 byte (number of collection migration filters)
 //	(per col filter) - 2 byte
-func (mut *Mutation) Serialize() []byte {
+func (mut *Mutation) Serialize(xattraIterator *xdcrBase.XattrIterator, XattrsToExclude map[string]bool) ([]byte, error) {
 	var bodyHash [64]byte
 	var xattrSize uint32
 	var xattr []byte
-
+	var bodyWithoutXattr, trimmedXattrPlusBody, hlv, importCas []byte
+	var err error
 	if mut.Datatype&xdcrBase.XattrDataType > 0 {
-		bodyWithoutXattr, err := xdcrBase.StripXattrAndGetBody(mut.Value)
+		bodyWithoutXattr, err = xdcrBase.StripXattrAndGetBody(mut.Value)
 		if err != nil {
-			fmt.Println("Error encountered while stripping Xattrs from the body, err: ", err)
+			return nil, err
 		}
-		bodyHash = sha512.Sum512(bodyWithoutXattr)
 		xattrSize, _ = xdcrBase.GetXattrSize(mut.Value)
 		xattr = mut.Value[4 : xattrSize+4]
+		trimmedXattrPlusBody, hlv, importCas, err = stripHlvAndImportFromXattr(xattr, len(mut.Value), xattrSize, xattraIterator, XattrsToExclude, bodyWithoutXattr)
+		if err != nil {
+			return nil, err
+		}
+		bodyHash = sha512.Sum512(trimmedXattrPlusBody)
```

Comment:
Can you double check that the xattrs are in sorted order?  You can check against KV team.

Is it possible that doc A has xattr in the order of <key1, value1> and <key2, value2>
Doc B has xattr <key2, value2>, <key1, value1>?

If the above is possible, is it possible then the SHA can be mismatched even though the document xattr is the same but just in different order?

Reply:
The KV team replied with, "Short answer is that you cannot depend on the ordering (if you try to fetch the SAME document multiple times the order would be the same, but if you do any modification to the document the ordering may or may not change)."
So i think we can stick to the current implementation of ordering the keys and then constructing the Xattrs based on the ordered keys.